### PR TITLE
Fixed Mocha for CoffeeScript 1.7+ and Travis CI errors

### PR DESCRIPTION
**Pull request by MrSaints**
`Gruntfile.js` and `test` are not currently compatible with CoffeeScript 1.7+ and they return the following error when executed:

```
Error: Use CoffeeScript.register() or require the coffee-script/register module to require .coffee.md files.
```

According to CoffeeScript's change log:

```
When requiring CoffeeScript files in Node you must now explicitly register the compiler. This can be done with require 'coffee-script/register' or CoffeeScript.register(). Also for configuration such as Mocha's, use coffee-script/register.
```

These changes fixes the error for when CoffeeScript 1.7+ is installed globally.

Also, https://github.com/hubot-scripts/hubot-example (which uses ~1.6) should probably be removed or edited to reflect the recent changes (Yeoman generator).

### CODE REVIEW

**Comment by technicalpickles** [comment]
Thanks for digging into this. Working through https://github.com/github/hubot/pull/863 and running basically into the same problems.

File: Gruntfile.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = function(grunt) {
       test: {
         options: {
           reporter: 'spec',
-          require: 'coffee-script'
+          require: 'coffee-script/register'
```
Comment (technicalpickles) [inline Gruntfile.js]: `package.json` still has coffee-script@1.6.3 for development. So, probably need to bump that, or revert this change.

File: generators/script/templates/.travis.yml [hunk 2]
```diff
@@ -3,4 +3,6 @@ node_js:
   - "0.11"
   - "0.10"
 notifications:
-  email: false
\ No newline at end of file
+  email: false
+before_script:
+  - npm install coffee-script grunt-cli
```
Comment (technicalpickles) [inline generators/script/templates/.travis.yml]: Do we get failures without this?

**Comment by technicalpickles** [comment]
I'm trying to think through this, and I'm having a hard time if scripts generated this will work with hubot as it is now, or if there needs to be support for newer coffeescript (ie https://github.com/github/hubot/pull/863)

**Comment by technicalpickles** [comment]
I tried generating a new script, then running `grunt test`, and got:

```
$ grunt test
Running "mochaTest:test" (mochaTest) task
Warning: Cannot find module 'coffee-script/register' Use --force to continue.

Aborted due to warnings.
```

Looks like coffee-script needs to be a dependency (or dev dependency) of the generated script.

File: generators/script/templates/.travis.yml (hunk 2 above)
Comment (MrSaints) [inline generators/script/templates/.travis.yml]: Yes.

About 2 weeks ago I enabled Travis CI for all my scripts. On Node 0.11, I seemed to be getting errors related to Grunt not being found, and on 0.10, it was the: `Warning: Cannot find module 'coffee-script/register' Use --force to continue.` error. Installing both of them before running the tests seemed to resolve the problem for me.

**Comment by MrSaints** [comment]
I agree. You might have to add the latest CoffeeScript as a dependency (or at least a dev dependency) in the script template. I realised that it doesn't work too well even with CoffeeScript installed globally if it is done via a NVM environment. It works best with a local copy of CoffeeScript in its directory.

File: Gruntfile.js (hunk 1 above)
Comment (MrSaints) [inline Gruntfile.js]: If we do revert this, we may be locking all future Hubot scripts to < CoffeeScript 1.7. I'm not sure what's the best way to  go about this (perhaps some condition for Travis?), but I personally feel that we should just bump it as it should only affect newly generated scripts.

**Comment by mtsmfm** [comment]
Hi, I noticed that generator-hubot generates package.json which references very old coffeescirpt.

How is the progress?

https://github.com/github/generator-hubot/blob/v0.4.0/generators/script/templates/_package.json#L29

My package (hubot-test-helper) user want to use "yield".
https://github.com/mtsmfm/hubot-test-helper/issues/41#issuecomment-275914732

**Comment by bkeepers** [comment]
We will be updating all of Hubot to plain JavaScript soon, so this issue should go away. https://github.com/hubotio/evolution/pull/4
